py3status/modules/external_script.py

```diff
@@ -96,10 +178,28 @@ def external_script(self):
         else:
             output = ''
 
+        self.script_data = {
+            'line': len(output_lines),
+            'output': output,
+            'output_full': ' '.join(output_full.splitlines())
+        }
         response['full_text'] = self.py3.safe_format(
-            self.format, {'output': output})
+            self.format, self.script_data
+        )
+
+        self.script_data['output_full'] = output_full
+        if self.notification['normal']:
+            self._notify_user()
+
         return response
 
+    def on_click(self, event):
+        button = event["button"]
+        if button != self.button_refresh:
+            if self.notification['click']:
```

Comment:
Just a reminder for future self: this throws exception if `notification` object doesn't contain `click` key

Reply:
What configs? I'm not able to reproduce. The `notification` dict should already have all three.

https://github.com/ultrabug/py3status/blob/c6a1c2146294ed1c90ba7153df3c9fb41d130fe9/py3status/modules/external_script.py#L111-L114

Reply:
This is my exact config:

```
order += 'external_script date'

external_script date {
  cache_timeout = 60
  script_path = 'date +"%a, %d %b"'
  format = 'Calendar: {output}'
  on_click 1 = 'exec gsimplecal'
}
```

Add this, then restart i3 (important), then click on the module.

I get an error, and in `journalctl` I see

```
py3status[12014]: on_click event in `external_script date` failed (KeyError) external_script.py line 199.
py3status[12014]: on_click event in `external_script date` failed (KeyError) external_script.py line 199. Please try to fix this and reload i3wm (Mod+Shift+R)
```